d2/src/main/java/com/linkedin/d2/balancer/dualread/DualReadStateManager.java

```diff
@@ -191,7 +191,14 @@ public void checkAndSwitchMode(String d2ServiceName)
 
     _executorService.execute(() ->
     {
-      boolean shouldCheck = _rateLimiter.tryAcquire();
+      if(d2ServiceName == null){
+        return;
+      }
```

Comment:
You should move this check outside of `_executorService`, there's no point in scheduling this if it won't run :) 
Also, watch the formatting!

Reply:
Thanks. Updated~